### PR TITLE
debug vectors and matrices using digits

### PR DESCRIPTION
**Pull request by patriciogonzalezvivo**
* Draw Vector digits
Screenshot from 2024-10-18 13-40-00

* Draw Matrix digits
image

### CODE REVIEW

File: draw/digits.hlsl [hunk 1]
```diff
@@ -118,7 +123,77 @@ float digits(in float2 st, in float value, in float nDecDigit, in float nIntDigi
...
+        rta += digits( pos, value );
+    }
+    return rta;
+}
+
+float digits(in float2 st, in float2x2 _matrix) {
+    float rta = 0.0;
+    for (int i = 0; i < 2; i++) {
+        for (int j = 0; j < 2; j++) {
+            float2 pos = st + float2(float(i), float(j)) * DIGITS_SIZE * DEBUG_VALUE_OFFSET - DIGITS_SIZE * vec2(0.0, 3.0);
```
Comment (shadielhajj) [inline draw/digits.hlsl:168]: vec2 should be float2

File: draw/digits.hlsl [hunk 2]
```diff
@@ -118,7 +123,77 @@ float digits(in float2 st, in float value, in float nDecDigit, in float nIntDigi
...
+        }
+    }
+    return rta;
+}
+
+float digits(in float2 st, in float3x3 _matrix) {
+    float rta = 0.0;
+    for (int i = 0; i < 3; i++) {
+        for (int j = 0; j < 3; j++) {
+            float2 pos = st + float2(float(i), float(j)) * DIGITS_SIZE * DEBUG_VALUE_OFFSET - DIGITS_SIZE * vec2(0.0, 6.0);
```
Comment (shadielhajj) [inline draw/digits.hlsl:180]: vec2 should be float2

File: draw/digits.hlsl [hunk 3]
```diff
@@ -118,7 +123,77 @@ float digits(in float2 st, in float value, in float nDecDigit, in float nIntDigi
...
+        }
+    }
+    return rta;
+}
+
+float digits(in float2 st, in float4x4 _matrix) {
+    float rta = 0.0;
+    for (int i = 0; i < 4; i++) {
+        for (int j = 0; j < 4; j++) {
+            float2 pos = st + float2(float(i), float(j)) * DIGITS_SIZE * DEBUG_VALUE_OFFSET - DIGITS_SIZE * vec2(0.0, 9.0);
```
Comment (shadielhajj) [inline draw/digits.hlsl:192]: vec2 should be float2

File: draw/digits.msl [hunk 4]
```diff
@@ -124,7 +128,78 @@ float digits(float2 st, float value, float nDecDigit, float nIntDigits) {
...
+        rta += digits( pos, value );
+    }
+    return rta;
+}
+
+float digits(in float2 st, in float2x2 _matrix) {
+    float rta = 0.0;
+    for (int i = 0; i < 2; i++) {
+        for (int j = 0; j < 2; j++) {
+            float2 pos = st + float2(float(i), float(j)) * DIGITS_SIZE * DEBUG_VALUE_OFFSET - DIGITS_SIZE * vec2(0.0, 3.0);
```
Comment (shadielhajj) [inline draw/digits.msl:173]: vec2 should be float2

File: draw/digits.msl [hunk 5]
```diff
@@ -124,7 +128,78 @@ float digits(float2 st, float value, float nDecDigit, float nIntDigits) {
...
+        }
+    }
+    return rta;
+}
+
+float digits(in float2 st, in float3x3 _matrix) {
+    float rta = 0.0;
+    for (int i = 0; i < 3; i++) {
+        for (int j = 0; j < 3; j++) {
+            float2 pos = st + float2(float(i), float(j)) * DIGITS_SIZE * DEBUG_VALUE_OFFSET - DIGITS_SIZE * vec2(0.0, 6.0);
```
Comment (shadielhajj) [inline draw/digits.msl:185]: vec2 should be float2

File: draw/digits.msl [hunk 6]
```diff
@@ -124,7 +128,78 @@ float digits(float2 st, float value, float nDecDigit, float nIntDigits) {
...
+        }
+    }
+    return rta;
+}
+
+float digits(in float2 st, in float4x4 _matrix) {
+    float rta = 0.0;
+    for (int i = 0; i < 4; i++) {
+        for (int j = 0; j < 4; j++) {
+            float2 pos = st + float2(float(i), float(j)) * DIGITS_SIZE * DEBUG_VALUE_OFFSET - DIGITS_SIZE * vec2(0.0, 9.0);
```
Comment (shadielhajj) [inline draw/digits.msl:197]: vec2 should be float2

**Comment by patriciogonzalezvivo** [comment]
thank you @shadielhajj

File: draw/digits.glsl [hunk 7]
```diff
@@ -124,7 +128,77 @@ float digits(in vec2 st, in float value, in float nDecDigit, in float nIntDigits
...
+    float rta = 0.0;
+    for (int i = 0; i < 3; i++) {
+        vec2 pos = st + vec2(float(i), 0.0) * DIGITS_SIZE * DEBUG_VALUE_OFFSET;
+        float value = i == 0 ? v.x : i == 1 ? v.y : v.z;
+        rta += digits( pos, value );
+    }
+    return rta;
+}
+
+float digits(in vec2 st, in vec4 v) {
```
Comment (shadielhajj) [inline draw/digits.glsl:159]: Should be float4 instead of vec4

File: draw/digits.hlsl [hunk 8]
```diff
@@ -118,7 +123,77 @@ float digits(in float2 st, in float value, in float nDecDigit, in float nIntDigi
...
+    float rta = 0.0;
+    for (int i = 0; i < 3; i++) {
+        float2 pos = st + float2(float(i), 0.0) * DIGITS_SIZE * DEBUG_VALUE_OFFSET;
+        float value = i == 0 ? v.x : i == 1 ? v.y : v.z;
+        rta += digits( pos, value );
+    }
+    return rta;
+}
+
+float digits(in float2 st, in vec4 v) {
```
Comment (shadielhajj) [inline draw/digits.hlsl:154]: Should be float4 instead of vec4

**Comment by shadielhajj** [comment]
Looks good to me @patriciogonzalezvivo! Very useful stuff.

**Review by shadielhajj — APPROVED** [review]

**Comment by patriciogonzalezvivo** [comment]
thank you @shadielhajj

**Pull request merged**